client/data/site-profiler/use-domain-analyzer-query.ts

```diff
@@ -0,0 +1,25 @@
+import { SourceSiteMigrationDetails } from '@automattic/data-stores/src/site';
+import { useQuery } from '@tanstack/react-query';
+import wp from 'calypso/lib/wp';
+
+export interface MigrationStatusError {
+	status: number;
+	message: string;
+}
+
+export const useDomainAnalyzerQuery = ( domain: string ) => {
+	return useQuery( {
+		queryKey: [ 'domain-', domain ],
+		queryFn: (): Promise< SourceSiteMigrationDetails > =>
```

Comment:
We can create a base response type for this first bootstrap. This one is enough for now.

```ts
{
  domain: string;
  hosting_provider: {
    name: string;
  };
  whois: any;
  dns: any;
}
```
We can then create a more detailed one.

Reply:
Good catch, this is copy/paste (SourceSiteMigrationDetails) momentum, and I forgot to update the response type.